Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -114,12 +116,13 @@ public Task<Page> PopAsync()
 
 		public async Task<Page> PopAsync(bool animated)
 		{
+			ActiveNavigationTaskCount++;
 			if (CurrentNavigationTask != null && !CurrentNavigationTask.IsCompleted)
 			{
 				var tcs = new TaskCompletionSource<bool>();
 				Task oldTask = CurrentNavigationTask;
 				CurrentNavigationTask = tcs.Task;
-				await oldTask;
+				await oldTask.ContinueWith((t) => ResolveActiveNavigation(t));
```

Comment:
I think the signature can change to passing `oldTask` to a method and awaiting the method instead. Similar pattern can be observed in other cases.